src/components/exchange/ExchangeTokenRow.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import isEqual from 'react-fast-compare';
+import { Box, Column, Columns, Inline, Stack, Text } from '@/design-system';
+import { isNativeAsset } from '@/handlers/assets';
+import { Network } from '@/helpers';
+import { useAccountAsset, useDimensions } from '@/hooks';
+import { ethereumUtils } from '@/utils';
+import FastCoinIcon from '../asset-list/RecyclerAssetList2/FastComponents/FastCoinIcon';
+import { ButtonPressAnimation } from '../animations';
+import { FloatingEmojis } from '../floating-emojis';
+import { IS_IOS } from '@/env';
+import {
+  FavStar,
+  Info,
+} from '../asset-list/RecyclerAssetList2/FastComponents/FastCurrencySelectionRow';
+
+interface ExchangeTokenRowProps {
+  item: any;
+}
+
+export default React.memo(function ExchangeTokenRow({
+  item: {
+    uniqueId,
+    showBalance,
+    showFavoriteButton,
+    showAddButton,
+    onPress,
+    theme,
+    nativeCurrency,
+    nativeCurrencySymbol,
+    favorite,
+    toggleFavorite,
+    contextMenuProps,
+    symbol,
+    address,
+    mainnet_address,
+    name,
+    testID,
+    type,
+    disabled,
+  },
+}: ExchangeTokenRowProps) {
+  const { width: deviceWidth } = useDimensions();
+
+  // TODO https://github.com/rainbow-me/rainbow/pull/3313/files#r876259954
+  const item = useAccountAsset(uniqueId, nativeCurrency);
+  const network = ethereumUtils.getNetworkFromType(type) ?? Network.mainnet;
+  const rowTestID = `${testID}-exchange-coin-row-${
+    symbol ?? item?.symbol ?? ''
+  }-${type || 'token'}`;
+
+  const isInfoButtonVisible =
+    !item?.isNativeAsset ||
+    (!isNativeAsset(address ?? item?.address, network) && !showBalance);
+
+  return (
+    <Columns alignVertical="center" space="10px">
+      <Column>
+        <Box
+          paddingLeft="20px"
+          as={ButtonPressAnimation}
+          // @ts-ignore
+          onPress={onPress}
```

Comment:
General question: `Box` doesn't accept `onPress`? Should it?

Reply:
I could be wrong but I think because this `Box` is cast `as={ButtonPressAnimation}`, it accepts `ButtonPressAnimation` props, but would be interested to know if that's not actually the case.

Reply:
Oh I just assumed the button took `onPress` too and that Box was the culprit, my bad!

Reply:
@derHowie you're correct but TS doesn't seem to like this pattern from what ive seen

@estrattonbailey I think the idea was to keep box as strictly a layout element, I dont have any strong opinions here